types/key/key.go

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) 2020 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package key defines some types related to curve25519 keys.
+package key
+
+import "golang.org/x/crypto/curve25519"
+
+// Private represents a curve25519 private key.
+type Private [32]byte
+
+// B32 returns k as the *[32]byte type that's used by the
+// golang.org/x/crypto packages.
+func (k *Private) B32() *[32]byte { return (*[32]byte)(k) }
```

Comment:
This aliases the memory of `k`. Do we want that? My instinct is to copy these tiny 4-register fields where possible to reduce the likelihood of something funny happening.

Reply:
I didn't want to change the behavior. The current code aliases them when calling box.Open/Seal, trusting box.* to not mess with them.

I could allocate instead, though. Peformance wise it probably won't matter yet, but it might in the future? How about I allocate for now and we can add fast aliasing versions in the future, or they can write the ugly conversion by hand if that's what the caller really wants.

Reply:
Done.